### PR TITLE
ArtifactCommandExtension changes

### PR DESCRIPTION
**Pull request by divyabattoju**
ArtifactCommandExtension changes

### CODE REVIEW

File: src/Agent.Worker/Variables.cs [hunk 1]
```diff
@@ -143,6 +143,16 @@ into maskHintGrouping
 
         public string Release_ReleaseUri => Get(Constants.Variables.Release.ReleaseUri);
 
+        public int? Release_ReleaseId => GetInt(Constants.Variables.Release.ReleaseId);
+
+        public int? Release_ReleaseEnvironmentId => GetInt("release.environmentId");
```
Comment (vithati) [inline src/Agent.Worker/Variables.cs]: use Constants.Variables

**Comment by bryanmacfarlane** [comment]
The title was changed from something useful "ArtifactCommandExtension changes for RM/Build convergence" to something not very useful "ArtifactCommandExtension changes".  What are these changes for?  Are they required for future RM/Build convergence?

**Comment by bryanmacfarlane** [comment]
Please hold.  The convergence cannot require agent changes.

File: src/Agent.Worker/Build/ArtifactCommandExtension.cs [hunk 2]
```diff
@@ -8,10 +8,166 @@
...
+            int? releaseEnvironmentId = context.Variables.Release_ReleaseEnvironmentId;
+            ArgUtil.NotNull(releaseId, nameof(releaseEnvironmentId));
+
+            int? attempt = context.Variables.Release_Attempt;
+            ArgUtil.NotNull(releaseId, nameof(attempt));
+
+            long? containerId = context.Variables.Release_ContainerId;
+            ArgUtil.NotNull(containerId, nameof(containerId));
+
+            ServiceEndpoint vssEndpoint = context.Endpoints.FirstOrDefault(e => string.Equals(e.Name, ServiceEndpoints.SystemVssConnection, StringComparison.OrdinalIgnoreCase));
```
Comment (omeshp) [inline src/Agent.Worker/Build/ArtifactCommandExtension.cs]: not required, connection is already passed in.

**Comment by TingluoHuang** [comment]
I don't think we need this change anymore.

**Pull request closed without merging**